Terminal.Gui/Core/Graphs/LineCanvas.cs

```diff
@@ -0,0 +1,520 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Terminal.Gui.Graphs {
+
+
+	/// <summary>
+	/// Facilitates box drawing and line intersection detection
+	/// and rendering.  Does not support diagonal lines.
+	/// </summary>
+	public class LineCanvas {
+
+		
+		private List<StraightLine> lines = new List<StraightLine> ();
+		private ConsoleDriver driver;
+
+		public LineCanvas (ConsoleDriver driver)
```

Comment:
> Is this puppy ready?

Yes its good to go.  I think it stands on its own and can be merged separately from the SplitContainer PR.

Reply:
In order to make API simpler for users (and because I can't think of a use case where you wouldn't just pass `Application.Driver` in anyway) I've removed the constructor.  In 8f41296 the code will just reference `Application.Driver` directly rather than taking it as a constructor arg.